tests/geobase/generate_TestInput.h

```diff
@@ -0,0 +1,175 @@
+        }
+    } else if (shape == "SPHE") {
+        constexpr int points{3};
+        for (int iPoint = 0; iPoint < points; ++iPoint) {
+            os << std::setw(9) << parameters.at(0).at(iPoint) << std::setw(10) << parameters.at(1).at(iPoint) << "\n";
+        }
+    } else if (shape == "PCON") {
+        int points = parameters.at(0).size();
+        for (int iPoint = 0; iPoint < points; ++iPoint) {
+            if (iPoint == 0) {
+                os << std::setw(3) << static_cast<Int_t>(parameters.at(0).at(iPoint)) << "\n";
+            } else if (iPoint == 1) {
+                os << std::setw(9) << parameters.at(0).at(iPoint) << std::setw(10) << parameters.at(1).at(iPoint)
+                   << "\n";
+            } else {
+                os << std::setw(9) << parameters.at(0).at(iPoint) << std::setw(10) << parameters.at(1).at(iPoint)
+                   << std::setw(10) << parameters.at(2).at(iPoint) << "\n";
+            }
+        }
+    } else if (shape == "PGON") {
+        int points = parameters.at(0).size();
+        for (int iPoint = 0; iPoint < points; ++iPoint) {
+            if (iPoint == 0) {
+                os << std::setw(3) << static_cast<Int_t>(parameters.at(0).at(iPoint)) << "\n";
+            } else {
+                os << std::setw(9) << parameters.at(0).at(iPoint) << std::setw(10) << parameters.at(1).at(iPoint)
+                   << std::setw(10) << parameters.at(2).at(iPoint) << "\n";
+            }
+        }
+    }
+}
+
+auto generateTempFileName()
+{
+    // generate a temporary file in an appropriate folder
+    // there is no good solution available from C and C++ standards
+    boost::filesystem::path tempDir = boost::filesystem::temp_directory_path();
+    boost::filesystem::path tempFile = boost::filesystem::unique_path();
+    return ( (tempDir / tempFile).native());
+}
+
+auto generateVolumeParameter(std::string shape, std::vector<std::vector<float>>& parameters)
+{
+
+    // stringstream which is used to collect all information added
+    // The stringstream is finally written to file
+    std::stringstream os;
+    os << std::fixed << std::setprecision(3);
+
+    fairroot::tests::getVolumeParameter(shape, parameters, os);
+
+    std::string outFileName = fairroot::tests::generateTempFileName();
+
+    std::ofstream outFile{};
+    outFile.open(outFileName);
+    outFile << os.rdbuf();
+    outFile.close();
+
+    return outFileName;
+}
+
+bool compareFiles(std::string file1, std::string file2)
+{
+    std::fstream f1, f2;
+    char c1, c2;
+
+    f1.open(file1.c_str(), std::ios::in);
+    if (!f1) {
+        std::cout << "File " << file1 << " can't be opened\n";
+        return false;
+    }
+
+    f2.open(file2.c_str(), std::ios::in);
+    if (!f2) {
+        f1.close();
+        std::cout << "File " << file2 << " can't be opened\n";
+        return false;
+    }
+
+    while (1) {
+        c1 = f1.get();
+        c2 = f2.get();
```

Comment:
get returns an integer. If you want to compare this with eof, then you probably should take the exact type that get returns.

```suggestion
        auto c1 = f1.get();
        auto c2 = f2.get();
```

I wonder whether there is something more elegant than reading character by character?

Reply:
If you have a better suggestion how to compare two files in C++ I am happy to change the code to a better version. For the time being I correct it as suggested.